orc8r/cloud/go/tools/swaggergen/generate/generate.go

```diff
@@ -242,6 +288,17 @@ func StripAndWriteSwaggerSpecs(specs map[string]MagmaSwaggerSpec) error {
 	return nil
 }
 
+// RetrieveServiceName parses the targetFilepath for its originating service.
+func RetrieveServiceName(targetFilePath string, rootDir string) (string, error) {
+	absTargetFilepath, err := filepath.Abs(targetFilePath)
+	if err != nil {
+		return "", errors.Wrapf(err, "target filepath %s is invalid", targetFilePath)
+	}
+	splitPath := strings.Split(absTargetFilepath, "/")
```

Comment:
https://golang.org/pkg/path/filepath/#SplitList

Reply:
I don't think SplitList wold be beneficial in this scenario since it works to split by List Separators (":") and not Path Separators ("/")

Reply:
Ah good catch. Okay then use strings.Split but with https://golang.org/pkg/os/#PathSeparator